server/http/helloapp/command.go

```diff
@@ -0,0 +1,153 @@
+		httputils.WriteJSON(w, apps.CallResponse{
+			Type:     apps.CallResponseTypeError,
+			Markdown: md.MD(errors.Wrap(err, msg).Error()),
+		})
+		return http.StatusInternalServerError, err
+	}
+
+	username := call.Values.Get("user")
+	message := call.Values.Get("message")
+	u, err := h.apps.Mattermost.User.GetByUsername(username)
+	if err != nil {
+		return e(err, "get by username "+username)
+	}
+
+	fullMessage := md.Markdownf("Hallo! Your teammate %s sent you a message: \"%s\"", call.Context.ActingUserID, message)
+	h.DM(u.Id, fullMessage.String())
+	h.Ephemeral(call.Context.ActingUserID, call.Context.ChannelID, fmt.Sprintf("Sent \"%s\" to @%s", message, username))
+	httputils.WriteJSON(w,
+		apps.CallResponse{
+			Type:     apps.CallResponseTypeOK,
+			Markdown: md.Markdownf("Looks like you've submitted the form!"),
+		})
+
+	return 200, nil
+}
+
+func (h *helloapp) handleHelloCommandDefinition(w http.ResponseWriter, req *http.Request, claims *apps.JWTClaims, call *apps.Call) (int, error) {
+	data := SubCommand{
+		Pretext: "/hello",
+		AppID:   AppID,
+		SubCommands: []SubCommand{
+			{
+				AppID:       AppID,
+				Pretext:     "sendto",
+				FormURL:     PathCommandSubmission,
+				Description: "Send a message to someone in this channel",
+				Args: []interface{}{
+					apps.AutocompleteDynamicSelect{
+						AutocompleteProps: apps.AutocompleteProps{
+							ElementProps: &apps.ElementProps{
+								Name:        "user",
+								Description: "User to send the message to",
+								Type:        apps.ElementTypeDynamicSelect,
+							},
+							AutocompleteElementProps: &apps.AutocompleteElementProps{
+								Hint:       "[username]",
+								Positional: true,
+							},
+						},
+						DynamicSelectElementProps: apps.DynamicSelectElementProps{
+							RefreshURL: "/lookup/users",
+						},
+					},
+					apps.AutocompleteText{
+						AutocompleteProps: apps.AutocompleteProps{
+							ElementProps: &apps.ElementProps{
+								Name:        "message",
+								Description: "The message to send",
+								Type:        apps.ElementTypeText,
+							},
+							AutocompleteElementProps: &apps.AutocompleteElementProps{
+								FlagName:   "message",
+								Positional: false,
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	res := apps.CallResponse{
+		Type: apps.CallResponseTypeOK,
+		Data: data,
+	}
+
+	httputils.WriteJSON(w, res)
+
+	return 200, nil
+}
+
+func (h *helloapp) handleLookupUsers(w http.ResponseWriter, req *http.Request, claims *apps.JWTClaims, call *apps.Call) (int, error) {
+	cmdStr := md.CodeBlock(call.Values.Raw)
+	cj := md.JSONBlock(call)
+
+	var e = func(err error) (int, error) {
+		h.DM(call.Context.ActingUserID, err.Error())
+		httputils.WriteJSON(w, apps.CallResponse{
+			Type:  apps.CallResponseTypeError,
+			Error: err,
+		})
+		return http.StatusInternalServerError, err
+	}
+
+	users, err := h.apps.Mattermost.User.ListInChannel(call.Context.ChannelID, "username", 0, 50)
```

Comment:
Do we always just want to look for users in the channel? I cannot run for example `/todo send @someone something` from a DM with the bot?

Reply:
This was just a demonstration to show that the context is being properly passed in and used. It's not really a practical feature of course.